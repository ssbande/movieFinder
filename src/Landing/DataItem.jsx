import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Card, Icon, Image } from 'semantic-ui-react';

import constants from '../utils/constants';
import { addDefaultSrc } from '../utils/helpers';

const DataItem = ({ item, handleCardClick }) => {
  const {
    id,
    poster_path,
    profile_path,
    title,
    original_title,
    original_name,
    name,
    release_date,
    first_air_date,
    vote_count
  } = item;

  return (
    <Grid.Column key={id} >
      <Card onClick={() => handleCardClick(id)} href="">
        <Image onError={addDefaultSrc} src={`${constants.imageBaseUrl}${poster_path || profile_path}`} alt={original_title} wrapped ui={false} />
        <Card.Content>
          <Card.Header className='breakWord'>{title || original_name || name}</Card.Header>
          <Card.Meta><Icon name='calendar alternate outline' /> {release_date || first_air_date}</Card.Meta>
        </Card.Content>
        <Card.Content extra>
          <div>
            <Icon name='thumbs up' />
            {vote_count}
          </div>
        </Card.Content>
      </Card>
    </Grid.Column>
  )
}

DataItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number.isRequired,
    poster_path: PropTypes.string,
    profile_path: PropTypes.string,
    title: PropTypes.string,
    original_title: PropTypes.string,
    original_name: PropTypes.string,
    name: PropTypes.string,
    release_date: PropTypes.string,
    first_air_date: PropTypes.string,
    vote_count: PropTypes.number
  }),
  handleCardClick: PropTypes.func.isRequired
}

export default DataItem;