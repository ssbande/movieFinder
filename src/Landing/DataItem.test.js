import React from 'react';
import { shallow } from 'enzyme';

import DataItem from './DataItem';

const setup = () => ({
  props: {
    item: {
      id: 1001,
      poster_path: 'poster_path',
      profile_path: 'profile_path',
      title: 'title1',
      original_title: 'original_title',
      original_name: 'original_name',
      name: 'name1',
      release_date: '2020-10-08',
      first_air_date: 'first_air_date',
      vote_count: 123
    },
    handleCardClick: jest.fn(),
  }
})

describe('data item', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<DataItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})