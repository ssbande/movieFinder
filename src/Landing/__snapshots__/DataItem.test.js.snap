// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data item should render correctly 1`] = `ShallowWrapper {}`;
